File: benchmarks/benchmark.js
// JS Benchmark: recursive fib(28) and loop of 100,000

function fib(n) {
  if (n <= 1) return n;
  return fib(n - 1) + fib(n - 2);
}

console.log("--- BENCHMARK JAVASCRIPT ---");

// 1. Fibonacci
const fibStart = performance.now();
const fibResult = fib(28);
const fibDuration = performance.now() - fibStart;
console.log(`1. Fibonacci(28) recursivo:`);
console.log(`   Resultado: ${fibResult}`);
console.log(`   Tiempo: ${fibDuration.toFixed(2)} ms`);

// 2. Loop
const loopStart = performance.now();
let count = 0;
for (let i = 0; i < 100000; i++) {
  count += 1;
}
const loopDuration = performance.now() - loopStart;
console.log(`2. Bucle simple de 100,000 iteraciones:`);
console.log(`   Resultado: ${count}`);
console.log(`   Tiempo: ${loopDuration.toFixed(2)} ms`);

File: benchmarks/benchmark.rs
// Rust Benchmark: recursive fib(28) and loop of 100,000
use std::time::Instant;

fn fib(n: u32) -> u32 {
    if n <= 1 {
        return n;
    }
    fib(n - 1) + fib(n - 2)
}

fn main() {
    println!("--- BENCHMARK RUST ---");

    // 1. Fibonacci
    let fib_start = Instant::now();
    let fib_result = fib(28);
    let fib_duration = fib_start.elapsed().as_secs_f64() * 1000.0;
    println!("1. Fibonacci(28) recursivo:");
    println!("   Resultado: {}", fib_result);
    println!("   Tiempo: {:.2} ms", fib_duration);

    // 2. Loop
    let loop_start = Instant::now();
    let mut count = 0;
    for _ in 0..100_000 {
        count += 1;
    }
    let loop_duration = loop_start.elapsed().as_secs_f64() * 1000.0;
    println!("2. Bucle simple de 100,000 iteraciones:");
    println!("   Resultado: {}", count);
    println!("   Tiempo: {:.2} ms", loop_duration);
}
